skglm/estimators.py

```diff
@@ -141,15 +143,26 @@ def _glm_fit(X, y, model, datafit, penalty):
     coefs, p_obj, kkt = solver(
         X_, y, datafit_jit, penalty_jit, w, Xw, max_iter=model.max_iter,
         max_epochs=model.max_epochs, p0=model.p0,
-        tol=model.tol,  # ws_strategy=model.ws_strategy,
+        tol=model.tol, fit_intercept=model.fit_intercept,
+        # ws_strategy=model.ws_strategy,
         verbose=model.verbose)
+    if y.ndim == 1:
+        model.coef_, model.stop_crit_ = coefs[:n_features], kkt
+        if model.fit_intercept:
+            model.intercept_ = coefs[-1]
+        else:
+            model.intercept_ = 0.
+    else:
+        model.coef_, model.stop_crit_ = coefs[:n_features, :], kkt
```

Comment:
you can do `coefs[:n_features]` and so this line becomes the same as for y.ndim==1 and they can be taken out of the if